src/app/clusters/bindings/bindings.cpp

```diff
@@ -120,55 +163,127 @@ bool emberAfBindingClusterBindCallback(app::CommandHandler * commandObj, const a
+                .clusterId  = entry.clusterId,
+            };
+            bindingTable.add(value);
+        }
+        else if (entry.type == EMBER_MULTICAST_BINDING && entry.fabricIndex == fabricIndex)
+        {
+            Structs::BindingEntry::Type value = {
+                .nodeId     = 0,
+                .groupId    = entry.groupId,
+                .endpointId = 0,
+                .clusterId  = entry.clusterId,
+            };
+            bindingTable.add(value);
+        }
     }
-    else
+    return encoder.EncodeList([&bindingTable](const auto & subEncoder) {
+        for (auto & value : bindingTable)
+        {
+            ReturnErrorOnFailure(subEncoder.Encode(value));
+        }
+        return CHIP_NO_ERROR;
+    });
+}
+
+CHIP_ERROR BindingTableAccess::Write(FabricIndex fabricIndex, const ConcreteDataAttributePath & path,
+                                     AttributeValueDecoder & decoder)
+{
+    switch (path.mAttributeId)
     {
-        bindingEntry = EmberBindingTableEntry::ForNode(fabricIndex, nodeId, localEndpoint, remoteEndpoint, clusterId);
+    case BindingList::Id:
+        return WriteBindingTable(fabricIndex, path.mEndpointId, decoder);
+    default:
+        break;
     }
+    return CHIP_NO_ERROR;
+}
 
-    uint8_t bindingIndex;
-    if (getBindingIndex(bindingEntry, &bindingIndex) != EMBER_SUCCESS)
+CHIP_ERROR BindingTableAccess::WriteBindingTable(FabricIndex fabricIndex, EndpointId endpoint, AttributeValueDecoder & decoder)
+{
+    BindingList::TypeInfo::DecodableType newBindingList;
+
+    ReturnErrorOnFailure(decoder.Decode(newBindingList));
+
+    // Add entries currently not in the binding table
+    auto iter      = newBindingList.begin();
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    uint8_t addedBindingIndecies[EMBER_BINDING_TABLE_SIZE];
+    uint8_t numAddedBindings = 0;
+    while (iter.Next())
     {
-        emberAfSendImmediateDefaultResponse(EMBER_ZCL_STATUS_NOT_FOUND);
-        return true;
+        if (!IsInBindingTable(fabricIndex, endpoint, iter.GetValue()))
+        {
+            err = AddBindingEntry(iter.GetValue(), fabricIndex, endpoint, &addedBindingIndecies[numAddedBindings]);
+            if (err != CHIP_NO_ERROR)
+            {
+                break;
+            }
+            numAddedBindings++;
+        }
     }
-
-    CHIP_ERROR err = BindingManager::GetInstance().UnicastBindingRemoved(bindingIndex);
+    // Revert the added entries upon error
     if (err != CHIP_NO_ERROR)
     {
-        ChipLogError(Zcl, "Binding: Failed to remove pending notification for unicast binding" ChipLogFormatX64 ": %s",
-                     ChipLogValueX64(nodeId), err.AsString());
+        for (uint8_t bindingIndex : addedBindingIndecies)
+        {
+            emberDeleteBinding(bindingIndex);
+        }
     }
 
-    emberDeleteBinding(bindingIndex);
-    emberAfSendImmediateDefaultResponse(EMBER_ZCL_STATUS_SUCCESS);
-    return true;
+    // Remove entries not in the new binding list
```

Comment:
Ignoring the ordering for the moment, this will completely fail if I try to overwrite the whole list with an equal length list, no?  If EMBER_BINDING_TABLE_SIZE == 16 and we already have 16 bindings and I try to write 16 new ones... we will try to add before we remove and it will just fail.